R/compare_performance.R

```diff
@@ -103,11 +119,29 @@ compare_performance <- function(..., metrics = "all", rank = FALSE, verbose = TR
   m <- mapply(function(.x, .y) {
     dat <- model_performance(.x, metrics = metrics, verbose = FALSE)
     model_name <- gsub("\"", "", .safe_deparse(.y), fixed = TRUE)
-    cbind(data.frame(Name = model_name, Model = class(.x)[1], stringsAsFactors = FALSE), dat)
+    perf_df <- data.frame(Name = model_name, Model = class(.x)[1], dat, stringsAsFactors = FALSE)
+    attributes(perf_df) <- c(attributes(perf_df), attributes(dat)[! names(attributes(dat)) %in% c("names", "row.names", "class")])
```

Comment:
`[! names(attributes(dat))` - here's a whitespace that could be removed.

Reply:
I can remove it, but I usually like to keep a space in `! x %in% y` constructions to avoid accidentally removing the `!` on later edits.